rust/src/db.rs

```diff
@@ -323,6 +324,36 @@ pub async fn load_payments() -> Result<Vec<PaymentInfo>> {
     Ok(payments)
 }
 
+/// Update status of an expired lightning payments,
+/// as LDK does not handle monitoring for expired ones.
+pub async fn clean_expired_payments() -> Result<()> {
+    let newly_expired = load_payments()
+        .await?
+        .iter()
+        // only act on pending payments
+        .filter(|info| info.status == HTLCStatus::Pending)
+        // check whether we have reached expiry time
+        .filter(|pending| {
+            pending
+                .expiry_timestamp
+                .and_then(|expiry| {
+                    if get_timestamp() > expiry {
+                        Some(expiry)
+                    } else {
+                        None
+                    }
+                })
+                .is_some()
+        })
+        .cloned()
+        .collect::<Vec<PaymentInfo>>();
+
+    for payment in newly_expired {
+        update_payment(&payment.hash, HTLCStatus::Failed, None).await?;
```

Comment:
🙃 It could be nice to differentiate between failed payments and expired payments in the UI.

Reply:
I have added capability for this in my changeset :-)- you can diferrentiate it the payment details (a new icon can be added with a one-liner in Dart after this gets in, I just didn't come up with any)